src/pkg/provisioner/disk_layout.go

```diff
@@ -151,6 +153,36 @@ func calcSDA3End(device string) (uint64, error) {
 	return sda3End, nil
 }
 
+func waitForDiskResize(deps Deps, runState *state) error {
+	if !runState.data.Config.BootDisk.WaitForDiskResize {
+		log.Println("WaitForDiskResize is not set, not waiting for a boot disk resize")
+		return nil
+	}
+	if runState.data.DiskResizeComplete {
+		log.Println("Already finished waiting for disk resize, not waiting again")
+		return nil
+	}
+	startSize, err := ioutil.ReadFile(filepath.Join(deps.RootDir, "sys/class/block/sda/size"))
+	if err != nil {
+		return err
+	}
+	log.Println("WaitForDiskResize is set; waiting for the boot disk size to change. Timeout is 3 minutes")
+	start := time.Now()
+	end := start.Add(3 * time.Minute)
+	for time.Now().Before(end) {
+		curSize, err := ioutil.ReadFile(filepath.Join(deps.RootDir, "sys/class/block/sda/size"))
+		if err != nil {
+			return err
+		}
+		if string(curSize) != string(startSize) {
+			log.Printf("Boot disk size has changed: start %q, end %q", strings.TrimSpace(string(startSize)), strings.TrimSpace(string(curSize)))
+			runState.data.DiskResizeComplete = true
+			return runState.write()
+		}
```

Comment:
would it be beneficial to have some sleep in between this waiting loop to keep cpu util in check

Reply:
Done